cpp/src/arrow/json/from_string.h

```diff
@@ -68,52 +68,45 @@ Result<std::shared_ptr<Array>> ArrayFromJSONString(const std::shared_ptr<DataTyp
 /// \brief Create a ChunkedArray from a JSON string
 ///
 /// \code {.cpp}
-/// std::shared_ptr<ChunkedArray> chunked_array;
-/// ChunkedArrayFromJSONString(
-///   int64(), {R"([5, 10])", R"([null])", R"([16])"}, &chunked_array
-/// );
+/// std::shared_ptr<ChunkedArray> chunked_array =
+///     ChunkedArrayFromJSONString(int64(), {R"([5, 10])", R"([null])", R"([16])"})
+///         .ValueOrDie();
```

Comment:
In general, error should be handled instead of using `ValueOrDie()`. So it may be better that we don't use `ValueOrDie()` in examples:

```suggestion
/// Result<std::shared_ptr<ChunkedArray>> chunked_array_result =
///     ChunkedArrayFromJSONString(int64(), {R"([5, 10])", R"([null])", R"([16])"});
```